easybuild/framework/easyconfig/types.py

```diff
@@ -467,33 +467,39 @@ def to_dependencies(dep_list):
     return [to_dependency(dep) for dep in dep_list]
 
 
+def _to_checksum(checksum, top_level=True, is_dict_value=False):
+    """Ensure the correct element type for each checksum in the checksum list"""
+    # each entry can be:
+    # * None (indicates no checksum)
+    # * a string (MD5, SHA256, ... checksum)
+    # * a list or tuple with 2 elements: checksum type + checksum value
+    # * a list or tuple of checksums (i.e. multiple checksums for a single file)
+    # * a dict (filename to checksum mapping)
+    if checksum is None or isinstance(checksum, string_type):
+        return checksum
+    elif isinstance(checksum, (list, tuple)):
+        if len(checksum) == 2 and isinstance(checksum[0], string_type) and isinstance(checksum[1], (string_type, int)):
+            # 2 elements => a checksum tuple (2nd element string or int) or 2 alternative checksums
+            return tuple(checksum)
```

Comment:
Does this only allow 2 "multiple checksums"? There could very well be more than 2 when pipy.org messes up things repeatedly